server/app/durablejobs/jobs/MigratePrimaryApplicantInfoJob.java

```diff
@@ -0,0 +1,111 @@
+import models.PersistedDurableJobModel;
+import repository.AccountRepository;
+import services.WellKnownPaths;
+import services.applicant.ApplicantData;
+import services.settings.SettingsService;
+
+public final class MigratePrimaryApplicantInfoJob extends DurableJob {
+
+  private final PersistedDurableJobModel persistedDurableJob;
+  private final AccountRepository accountRepository;
+  private final SettingsService settingsService;
+  private final Config config;
+
+  public MigratePrimaryApplicantInfoJob(
+      PersistedDurableJobModel persistedDurableJob,
+      AccountRepository accountRepository,
+      SettingsService settingsService,
+      Config config) {
+    this.persistedDurableJob = checkNotNull(persistedDurableJob);
+    this.accountRepository = checkNotNull(accountRepository);
+    this.settingsService = checkNotNull(settingsService);
+    this.config = checkNotNull(config);
+  }
+
+  @Override
+  public PersistedDurableJobModel getPersistedDurableJob() {
+    return persistedDurableJob;
+  }
+
+  @Override
+  public void run() {
+    /*
+     * In this job, we always copy data from the Well Known Path (WKP) to the Primary Applicant Info (PAI)
+     * columns, if the WKP data is present.  Data in the PAI columns will get overwritten. This is so that
+     * if a preseeded question is answered while the feature flag is off (or in the case of DOB, a
+     * question whose name happens to match where we are putting the data from TI client creation), that
+     * data gets synced over to the PAI columns.  The exception is for email address, for which there is
+     * no WKP. We will copy the account email over to the PAI column only if the PAI column is empty.
+     *
+     * When updating a TI client, both PAI and WKP get updated with the new data, so the two locations
+     * should remain in sync.
+     *
+     * When the flag is on, this job will be a no-op, as we will be removing WKP and PAI
+     * should be the authoritative source of truth for the data about the applicant. Because the flag
+     * will be ADMIN_WRITEABLE at first, then ADMIN_READABLE when we want to turn it on for everybody,
+     * and we don't have access to a request object, we need to check both the database and the config
+     * file setting to determine if the flag is on.
+     */
+    boolean doMigration;
+    Optional<ImmutableMap<String, String>> writeableSettings =
+        settingsService.loadSettings().toCompletableFuture().join();
+    if (writeableSettings.isPresent()
+        && writeableSettings.get().containsKey("PRIMARY_APPLICANT_INFO_QUESTIONS_ENABLED")) {
+      doMigration =
+          !writeableSettings.get().get("PRIMARY_APPLICANT_INFO_QUESTIONS_ENABLED").equals("true");
+    } else {
+      try {
+        doMigration = !config.getString("primary_applicant_info_questions_enabled").equals("true");
+      } catch (ConfigException.Missing e) {
+        doMigration = false;
+      }
+    }
+    if (doMigration) {
+      accountRepository
```

Comment:
suggestion (non-blocking): Add an info level log here to say that the data was migrated.